### ISSUE TITLE
Support selecting a video resolution and/or framerate

### ISSUE THREAD

**Issue by djee-ms** [microsoft/MixedReality-WebRTC#46]
The current video interface supports selecting a preferred video capture device and video codec, but not specifying a preferred resolution and/or framerate for the capture stream.

**Comment by eirikhollis** [comment]
Related to the resolution, is it possible to change that in _api.cpp_?
Thinking about lines 453 - 459, specifically (you can see I've already changed it):
```
453     cricket::VideoFormat max_format{};
454     max_format.width = 1280;
455     max_format.height = 720;
456     max_format.interval = cricket::VideoFormat::FpsToInterval(15);
457     //video_capturer->set_enable_camera_list(
458     //   true);  //< needed to enable filtering
459     video_capturer->ConstrainSupportedFormats(max_format);
```
Uncommenting out 457-458 seems to cause exceptions on HL1.

**Comment by djee-ms** [comment]
Several remarks on this:
- I never really managed to make that work, hence why this is commented and the feature not yet implemented. There are also issues with the internal filtering only taking into account the resolution and not the framerate, making that even more confusing.
- It was my understanding that media constraints were deprecated. Though I cannot find again any link to the issue/discussion where I saw that, and now I am doubting whether I misunderstood or misremembered it. 😕
- In any case, these are not media constraints, these are direct `cricket` constraints. Explicit references to the `cricket` video capture subsystem are removed from the API in m75 in favor of a clean abstraction of the video capture module, so that would be a temporary solution anyway.

That being said for the time being this is probably the way to go on Desktop.

For UWP, the video profile change from `feature/video_profile` will bring support for resolution and framerate selection through video profiles. I am waiting for the associated changes to land upstream; see webrtc-uwp/webrtc-apis#24.

**Comment by eirikhollis** [comment]
The reason I mentioned it is because I don't seem to get it working neither. It defaults to 30fps and the resolution seems to change during the connection? Probably dependant on the bitrate then?

Mediaconstraints are removed in m75, but still available in m71 as far as I know.

I've messaged you on holodevelopers slack if you're available there.

**Comment by djee-ms** [comment]
This is now fixed, with video capture formats (including resolution) available via media constraints (see a2a98554e59ee3b1757806670c3ccda8fc4d2464) and also via video profiles on UWP (see #67).
